pkg/infra/metrics/frontendmetrics.go

```diff
@@ -0,0 +1,43 @@
+package metrics
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// PostPostFrontendMetricsCommand sent by frontend to record frontend metrics
+type PostFrontendMetricsCommand struct {
+	Events []FrontendMetricEvent `json:"events"`
+}
+
+// FrontendMetricEvent a single metric measurement event
+type FrontendMetricEvent struct {
+	Name  string  `json:"name"`
+	Value float64 `json:"value"`
+}
+
+// FrontendMetricsRecorder handles the recording of the event, ie passes it to a prometheus metric
+type FrontendMetricsRecorder func(event FrontendMetricEvent)
+
+// MFrontendLoadTime is a metric summary of alert execution duration
+var FrontendMetrics map[string]FrontendMetricsRecorder = map[string]FrontendMetricsRecorder{}
+
+func registerFrontendSummary(name string, help string) {
+	objectiveMap := map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001}
+
+	summary := prometheus.NewSummary(prometheus.SummaryOpts{
+		Name:       name,
+		Help:       help,
+		Objectives: objectiveMap,
+		Namespace:  ExporterName,
+	})
+
+	FrontendMetrics[name] = func(event FrontendMetricEvent) {
+		summary.Observe(event.Value)
+	}
+
+	prometheus.MustRegister(summary)
+}
+
+func initFrontendMetrics() {
+	registerFrontendSummary("frontend_boot_load_time_milliseconds", "Frontend boot time measurement")
```

Comment:
```suggestion
	registerFrontendSummary("frontend_boot_load_time_seconds", "Frontend boot time measurement")
```

Reply:
Prometheus monitoring defaults to SI units so durations should be in seconds. Since it's a float64 we can get MS precision if we want anyway.

Reply:
Perhaps the frontend can measure in ms and we convert it to seconds https://github.com/grafana/grafana/pull/32593/files#diff-d633e21cdbb1603ed89c5abac34b4d816dc9932fac50ea550af81924eb58011eR33 here to make things easy in frontend and backend?

Reply:
I took this from our other metrics and they where named time_milliseconds